packages/contracts/contracts/L1/rollup/CanonicalTransactionChain.sol

```diff
@@ -516,11 +476,9 @@ contract CanonicalTransactionChain is ICanonicalTransactionChain, Lib_AddressRes
             blockNumber = lastElement.blockNumber;
         }
 
-        // For efficiency reasons getMerkleRoot modifies the `leaves` argument in place
-        // while calculating the root hash therefore any arguments passed to it must not
-        // be used again afterwards
+        // Cache the previous blockhash to ensure all transaction data can be retrieved efficiently.
         _appendBatch(
-            Lib_MerkleTree.getMerkleRoot(leaves),
+            blockhash(block.number-1),
```

Comment:
This is `block.number - 1`?

Reply:
Best we can do. Can't access the current hash at runtime

Reply:
This should be tested actually.

Reply:
What's the point of storing this block hash? Might as well just write a script that SSTOREs the block hash several times a day.

Reply:
@smartcontracts it's just easier to store at the time of submission at this point while we don't have the service. Long term we definitely want 1 single L1 block hash oracle which all rollups use so we never need to store a block hash here again. But until that point, it doesn't hurt to hash it into what we already store. If you'd like we can technically remove the storage entirely and have no cache of block hashes, but it's pretty minimal overhead at this point so figured we might as well keep it just in case.